admin/stats_products_purchased.php

```diff
@@ -6,257 +6,223 @@
-                <td class="dataTableHeadingContent"><?php echo TABLE_HEADING_ORDERS_DATE_PURCHASED; ?></td>
-                <td class="dataTableHeadingContent"><?php echo TABLE_HEADING_CUSTOMERS_INFO; ?></td>
-                <td class="dataTableHeadingContent" align="center"><?php echo TABLE_HEADING_PRODUCTS_QUANTITY; ?>&nbsp;</td>
-                <td class="dataTableHeadingContent" align="center"><?php echo TABLE_HEADING_PRODUCTS_NAME; ?>&nbsp;</td>
-                <td class="dataTableHeadingContent" align="center"><?php echo TABLE_HEADING_PRODUCTS_MODEL; ?>&nbsp;</td>
-              </tr>
-
-<?php
-  if ($chk_orders_products->EOF) {
-?>
+        $rows = 0;
+        $chk_orders_products = $db->Execute($chk_orders_products_query);
+        ?>
+        <table class="table table-hover">
+          <thead>
+            <tr class="dataTableHeadingRow">
+              <th class="dataTableHeadingContent"><?php echo TABLE_HEADING_CUSTOMERS_ID; ?></th>
+              <th class="dataTableHeadingContent"><?php echo TABLE_HEADING_ORDERS_ID; ?></th>
+              <th class="dataTableHeadingContent"><?php echo TABLE_HEADING_ORDERS_DATE_PURCHASED; ?></th>
+              <th class="dataTableHeadingContent"><?php echo TABLE_HEADING_CUSTOMERS_INFO; ?></th>
+              <th class="dataTableHeadingContent text-center"><?php echo TABLE_HEADING_PRODUCTS_QUANTITY; ?></th>
+              <th class="dataTableHeadingContent text-center"><?php echo TABLE_HEADING_PRODUCTS_NAME; ?></th>
+              <th class="dataTableHeadingContent text-center"><?php echo TABLE_HEADING_PRODUCTS_MODEL; ?></th>
+            </tr>
+          </thead>
+          <tbody>
+              <?php if ($chk_orders_products->EOF) { ?>
               <tr class="dataTableRowSelectedBot">
                 <td colspan="7" class="dataTableContent" align="center"><?php echo NONE; ?></td>
               </tr>
-<?php } ?>
-<?php
-  while (!$chk_orders_products->EOF) {
-    $rows++;
+            <?php } ?>
+            <?php
+            foreach ($chk_orders_products as $orders_products) {
+              $rows++;
 
-    if (strlen($rows) < 2) {
-      $rows = '0' . $rows;
-    }
-    if ($products_filter != '') {
-    // products_id
-      $cPath = zen_get_product_path($products_filter);
-    } else {
-    // products_name or products_model
-      $cPath = zen_get_product_path($chk_orders_products->fields['products_id']);
-    }
-?>
+              if (strlen($rows) < 2) {
+                $rows = '0' . $rows;
+              }
+              if ($products_filter != '') {
+                // products_id
+                $cPath = zen_get_product_path($products_filter);
+              } else {
+                // products_name or products_model
+                $cPath = zen_get_product_path($orders_products['products_id']);
+              }
+              $type_handler = $zc_products->get_admin_handler($product['products_type']);

```

Comment:
Is `$product` already defined here? And `$zc_products` too?

Reply:
I had to add that line, because $type_handler was undefined, and gave an error on the page

Reply:
I see $product defined in similar code later in 195, 202, 206, but not up here.

Reply:
I see `$zc_products` is part of the init autoloaders. So that part's fine.

The `$product` var is not. (The code may "work" because of fallbacks use to ensure a sane response is obtained.)

I think you'll need to change `$product['products_type']` to `zen_get_products_type($orders_products['products_id'])` in this case.

Reply:
And also on `202`